checkov/azure_pipelines/runner.py

```diff
@@ -30,3 +34,56 @@ def _parse_file(
 
     def is_workflow_file(self, file_path: str) -> bool:
         return file_path.endswith(('azure-pipelines.yml', 'azure-pipelines.yaml'))
+
+    def get_resource(self, file_path: str, key: str, supported_entities: Iterable[str],
+                     definitions: dict[str, Any] | None = None, root_folder: str | Path | None = None) -> str:
+        relative_file_path = f"/{os.path.relpath(file_path, root_folder)}"
+        if not self.definitions or not isinstance(self.definitions, dict):
+            return relative_file_path
+        start_line, end_line = Runner.get_start_and_end_lines(key)
+        resource_name = generate_resource_key_recursive(start_line, end_line, self.definitions[file_path])
+        if not resource_name:
+            return relative_file_path
+        return f"{relative_file_path}:{resource_name}"
+
+    @staticmethod
+    def get_start_and_end_lines(key: str) -> list[int]:
+        check_name = key.split('.')[-1]
+        if "[" not in check_name or "[]" in check_name:
+            return [-1, -1]
+
+        start_end_line_bracket_index = check_name.index('[')
+
+        return [int(x) for x in check_name[start_end_line_bracket_index + 1: len(check_name) - 1].split(':')]
```

Comment:
I think a few runners already have this functionality, and probably there will be more - wdyt about moving it to the object runner?

Reply:
Right!